tests/testthat/test-create_task_makefile.R

```diff
@@ -0,0 +1,110 @@
+context("create_task_makefile")
+
+task_config <- data.frame(
+  id=c('AZ','CO','CA'),
+  capital=c('Phoeniz','Denver','Sacramento')
+)
+step1 <- create_task_step(
+  step_name = 'prep',
+  target = function(task_name, step_name, ...) {
+    sprintf('%s_%s', task_name, step_name)
+  },
+  depends = c('A','B'),
+  command = "process(target_name, I('C'))"
+)
+step2 <- create_task_step(
+  step_name = 'plot',
+  command = function(target_name, task_name, ...) {
+    capital <- task_config[task_config$id == task_name, 'capital']
+    sprintf('visualize(\'%s\', \'%s\')', task_name, capital)
+  }
+)
+step3 <- create_task_step('report')
+task_plan <- create_task_plan(c('AZ','CA','CO'), list(step1, step2, step3),
+                              final_steps='report', ind_dir='states/log')
+
+test_that("can create task_plan", {
+
+  expect_true(all(names(task_plan) == c('AZ','CA','CO')))
+})
+
+test_that("can create task_makefile as string with defaults", {
+  task_makefile <- create_task_makefile(
+    task_plan, makefile=NULL,
+    file_extensions=c('ind'), packages=c('scipiper','mda.streams'))
+  expect_is(task_makefile, "character")
+  remake_list <- yaml::yaml.load(task_makefile)
+  expect_equal(tail(names(remake_list$targets), 1), "all_tasks.ind_promise")
+  expect_equal(head(names(remake_list$targets), 1), "all_tasks")
+  expect_equal(tail(remake_list$targets, 1)[[1]]$command, 
+               "combine_to_ind(I('states/log/all_tasks.ind'), 'states/log/AZ.ind', 'states/log/CA.ind', 'states/log/CO.ind')")
+})
+
+test_that("can create task_makefile with named target", {
+  task_makefile <- create_task_makefile(
+    task_plan, makefile=NULL, target_name = 'my_states',
+    file_extensions=c('ind'), packages=c('scipiper','mda.streams'))
+  remake_list <- yaml::yaml.load(task_makefile)
+  expect_equal(tail(names(remake_list$targets), 1), "my_states_promise")
```

Comment:
Geez, that was short-sighted of me. I like your idea, but it brings up a tricky part of the implementation. If we use the last arg as the named file path for `combine_to_ind` to pass `ind_file` to `sc_indicate`, we would then need to have consistency in arg naming of other combiner functions. Something like this:

```r
combine_to_ind(..., filepath){
  # bla bla
}
```
or more generic as
`combine_to_ind(..., x)`

My thinking on this originally (let me know if this no longer holds water) was to have two types of combiner functions, ones that write files and ones that return objects. The file writers take a file as the first arg and `...` for the rest of the objects/files to be part of the combiner operations. The object creators are the same, but have no named arg (only `...`). So me about 2 months ago would say, "well, `combine_to_ind` is a file writer combiner, and you are using it incorrectly, you must want `combine_to_hash`..." But the behavior you are pointing out above is kind of dangerous, because you would end up unknowingly re-writing the contents of 'states/log/AZ.ind' to reflect the combined hashes or the other two files...which isn't good.

Reply:
Just wrote four different replies to this thread. Here's my final response (for now :smile:):

For non-default finalizers, I think the user just has to be responsible for creating an appropriate match between their `final_targets` and `finalize_funs`, and I don't see any reliable ways for us to check on the user's behalf within `create_task_makefile`. Therefore there's no huge advantage to moving the ind_file argument to the end, and I agree that keeping it first has the nice property of allowing the user to define their own argument name for the output file.

However, `combine_to_ind` is special because it's the default. If `final_targets` is not supplied, the current code chooses a filename. I think it would be reasonable and useful for `create_task_makefile` to check and throw an error if `length(finalize_fun) == length(final_targets) == 1` and `finalize_fun=='combine_to_ind'` and yet `!remake:::target_is_file(final_targets, file_extensions = pos_extensions)`.

To specify a `final_target` that's an object, the user (and the tests) will therefore have to also specify a different `finalize_fun`, anything except `combine_to_ind`. I think that'd be fine/good.

Reply:
What do you think @jread-usgs - add a check with stop() in create_task_makefile and then modify the above test to expect or not trigger that stop()? Or did you already do this?

Reply:
Great ideas. I forgot this was left unanswered by me. I agree w/ your expectation for the defaults

Reply:
Instead of checking for length(final_targets) == 1, etc. I checked that the finalize_funs[i] wasn't `combine_to_ind` when the final_targets[i] was an object. This covers your case, but also a few other cases too and was simpler to implement and simpler for future people to understand the code at that spot.

Reply:
Yeah, nice solution!